### ISSUE TITLE
[content-service] ability to exclude content from the backup of `/workspace`

### ISSUE THREAD

**Issue by kylos101** [gitpod-io/gitpod#15500]
## Is your feature request related to a problem? Please describe

Sometimes users add large files to workspaces at runtime, use them, and then delete them, to avoid backing up the files to Gitpod.

Keeping the backup size smaller helps keep workspace restarts fast, because then we do not have to download and untar unnecessary files from object storage.

## Describe the behaviour you'd like

Provide a means to exclude file patterns from the backup of `/workspace`

## Describe alternatives you've considered

- Add the files to your `.gitignore` and delete them before stopping your workspace
- Implement a pre-stop task, and delete files as part of that task

## Additional context

`leeway run dev:preview` persists 20GB to the file system, but, we do not want to back it up, as an example

**Comment by kylos101** [comment]
@atduarte how would you like this experience to work? I can see us allowing users to exclude files from the backup via `.gitpod.yml`. Another option would be to allow users to define pre-stop tasks.

**Comment by svenefftinge** [comment]
I think this would be a good use case for a "stop" lifecycle hook.

**Comment by atduarte** [comment]
@svenefftinge indeed. I believe we should start there, as it is the mechanism that covers the most use cases and "makes sense".
